haunt-roll-fail/arcs/game-common.scala

```diff
@@ -890,11 +906,20 @@ object CommonExpansion extends Expansion {
 
         case BattleProcessAction(f, r, e, l, then) =>
             val sd = l.count(OwnDamage)
-            val ic = l.has(Intersept).??(e.at(r).ships.diff(e.damaged).num)
+            var ic = l.has(Intersept).??(e.at(r).ships.diff(e.damaged).num)
             val hs = l.count(HitShip)
             val bb = l.count(HitBuilding)
             val rd = l.count(RaidKey)
 
+            if (l.has(Intersept) && e.can(Irregular)) {
+                ic = e.resources.count(Weapon) + e.loyal.of[GuildCard].count(_.suit == Weapon)
+
+                if (e.resources.has(Weapon)) {
+                    e.remove(ResourceRef(Weapon, None))
+                    e.log("discarded", Weapon, "due to", Irregular)
+                }
+            }
+
```

Comment:
I feel this can be written more elegantly to integrate with L#909 better.

Reply:
When we do Mirror Plating and Signal Breaker we can address that.
The logic would be split between figuring out whether we intercept at all and then by how much.